tests/e2e/system_test/test_graceful_nodes_shutdown.py

```diff
@@ -0,0 +1,708 @@
+                self.second_bucket_name
+            )
+        }, "Needed uploaded objects could not be found"
+        compare_bucket_object_list(
+            mcg_obj_session, self.first_bucket_name, self.second_bucket_name
+        )
+
+    def validate_data_integrity(self):
+
+        logger.info("Compare bucket replication after reboot (unidirectional)")
+        assert compare_bucket_object_list(
+            self.mcg_obj_session, self.source_bucket_name, self.target_bucket_name
+        ), f"Failed on data validation for buckets {self.source_bucket_name} and {self.target_bucket_name}"
+
+        logger.info("Compare bucket replication after reboot (bidirectional)")
+        compare_bucket_object_list(
+            self.mcg_obj_session, self.first_bucket_name, self.second_bucket_name
+        )
+
+        logger.info("Compare bucket object after reboot")
+        for idx, bucket_name in enumerate(self.bucket_names_list):
+            after_bucket_object_set = {
+                obj.key
+                for obj in self.mcg_obj.s3_list_all_objects_in_bucket(bucket_name)
+            }
+            assert (
+                self.before_bucket_object_list[idx] == after_bucket_object_set
+            ), f"Data integrity failed for S3 bucket {bucket_name}"
+
+        logger.info("Compare PVCs MD5Sum after reboot")
+        assert self.ne_pvc_orig_md5_sum == pod.cal_md5sum(
+            self.ne_pvc_pod_obj, self.ne_file_name
+        ), (
+            f"Data integrity failed for file '{self.ne_file_name}' "
+            f"on non encrypted pvc {self.ne_pvc_obj.name} on pod {self.ne_pvc_pod_obj.name}"
+        )
+
+        assert self.eb_pvc_orig_md5_sum == pod.cal_md5sum(
+            self.eb_pvc_pod_obj, self.eb_file_name
+        ), (
+            f"Data integrity failed for file '{self.eb_file_name}' "
+            f"on encrypted block pvc {self.eb_pvc_obj.name} on pod {self.eb_pvc_pod_obj.name}"
+        )
+
+        assert self.efs_pvc_orig_md5_sum == pod.cal_md5sum(
+            self.efs_pvc_pod_obj, self.efs_file_name
+        ), (
+            f"Data integrity failed for file '{self.efs_file_name}' "
+            f"on encrypted fs pvc {self.efs_pvc_obj.name} on pod {self.efs_pvc_pod_obj.name}"
+        )
+
+    @system_test
+    @polarion_id("OCS-3976")
+    @skipif_no_kms
+    @skipif_ocs_version("<4.11")
+    def test_graceful_nodes_shutdown(
+        self, pgsql_factory_fixture, couchbase_factory_fixture, nodes
+    ):
+        """
+        Steps:
+          1) Have a cluster with FIPS, hugepages, encryption enabled
+          2) Create some resources: Create PVC (encrypt & non encrypt),
+            take snapshot and restore it into new PVCs, clone PVC
+          3) Configure OCP workloads(monitoring, registry, logging)
+          4) Configure PGSQL, Couchbase, AMQ , Cosbench, Jenkin, Quay workloads
+          5) Create rgw kafka notifications and objects should be notified
+          6) Create normal OBCs and buckets
```

Comment:
https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/longevity_helpers.py#L627 would it help?